src/components/wizards/create-vm/CreateVMWizard.js

```diff
@@ -0,0 +1,366 @@
+        type: 'dropdown',
+        default: '--- Select Provision Source ---',
+        values: [
+          {
+            name: 'PXE'
+          },
+          {
+            name: 'URL'
+          },
+          { 
+            name: 'Template'
+          },
+          { 
+            name: 'Registry'
+          }
+        ],
+        required: true
+      },
+      template: {
+        title: 'Template',
+        type: 'dropdown',
+        default: '--- Select Template ---',
+        required: true,
+        isVisible: (basicSettings) => _.get(basicSettings, 'imageSourceType.value') === 'Template',
+        labelSize: 4,
+        values: this.props.templates
+      },
+      registryImage: {
+        title: 'Registry Image',
+        required: true,
+        type: 'text',
+        labelSize: 4,
+        isVisible: (basicSettings) => _.get(basicSettings,'imageSourceType.value') === 'Registry'
+      },
+      imageURL: {
+        title: 'URL',
+        required: true,
+        type: 'text',
+        labelSize: 4,
+        isVisible: (basicSettings) => _.get(basicSettings,'imageSourceType.value') === 'URL'
+      },
+      operatingSystem: {
+        title: 'Operating System',
+        type: 'dropdown',
+        default: '--- Select Operating System ---',
+        values: this.props.operatingSystems,
+        required: true
+      },
+      flavor:{
+        title: 'Flavor',
+        type: 'dropdown',
+        default: '--- Select Flavor ---',
+        values: this.props.flavors.concat([{name:'Custom'}]),
+        required: true
+      },
+      memory:{
+        title: 'Memory (GB)',
+        type: 'text',
+        required: true,
+        isVisible: (basicSettings) => _.get(basicSettings, 'flavor.value', '') === 'Custom',
+        labelSize: 4
+      },
+      cpu: {
+        title: 'CPUs',
+        type: 'text',
+        required: true,
+        isVisible: (basicSettings) => _.get(basicSettings, 'flavor.value', '') === 'Custom',
+        labelSize: 4
+      },
+      workloadProfile:{
+        title: 'Workload Profile',
+        type: 'dropdown',
+        default: '--- Select Workload Profile ---',
+        values: this.props.workloadProfiles,
+        required: true
+      },
+      startVM:{
+        title: 'Start virtual machine on creation',
+        type: 'checkbox'
+      },
+      createTemplate:{
+        title: 'Create new teplate from configuration',
+        type: 'checkbox'
+      },
+      cloudInit:{
+        title: 'Use cloud-init',
+        type: 'checkbox',
+      },
+      hostName:{
+        title: 'Hostname',
+        type: 'text',
+        isVisible: (basicSettings) => _.get(basicSettings, 'cloudInit.value', false),
+        required: true
+      },
+      authKeys:{
+        title: 'Authenticated SSH Keys',
+        type: 'textarea',
+        isVisible: (basicSettings) => _.get(basicSettings, 'cloudInit.value', false),
+        required: true
+      }
+  
+    }
+  
+    wizardStepsNewVM = [
```

Comment:
I think the intent is to simplify the render by isolating wizard's steps into a separate variable (or technically an instance property).

We should strive to separate "what" from "how" so that the actual component is as simple as possible.

Reply:
makes sense, + extracting the generic stuff outside of this component

Reply:
thats why i wanted to start with redux, so you have to split it...